src/ln/channel.rs

```diff
@@ -934,7 +934,19 @@ impl Channel {
 			}, None));
 		}
 
-		transaction_utils::sort_outputs(&mut txouts);
+		transaction_utils::sort_outputs(&mut txouts, |a, b| {
+			if let &Some(ref a_htlc) = a {
+				if let &Some(ref b_htlc) = b {
+					a_htlc.0.cltv_expiry.cmp(&b_htlc.0.cltv_expiry)
+						// Note that due to hash collisions, we have to have a fallback comparison
+						// here for fuzztarget mode (otherwise at least chanmon_fail_consistency
+						// may fail)!
+						.then(a_htlc.0.payment_hash.0.cmp(&b_htlc.0.payment_hash.0))
```

Comment:
Hmm didn't get you there, if offered HTLC A and offered HTLC B have same pubkey script, isn't it implied that they have same payment hash, as per BOLT-3, hash is part of script and it's the only diff for HTLCs with same per commitment point ?

Reply:
Not in fuzztarget, in fuzztarget hashes are just (roughly) the whole input xor'd together, with 31 bytes being set to 0s. Thus, its rather trivial to have different payment hashes have the same script_pubkey.

Reply:
Okay make sense